File: scripts/dev.ts
#!/usr/bin/env -S deno run --allow-read --allow-run --allow-env

/**
 * Development script for resume project
 * Opens the PDF and starts typst watch
 */

import $ from "jsr:@david/dax@0.42.0";
import { dirname, join, basename } from "jsr:@std/path@1.0.8";

// Get the current working directory
const cwd = Deno.cwd();

// Change to the resume directory (parent of scripts) only if currently in scripts dir
if (basename(cwd) === "scripts") {
  const scriptPath = import.meta.url.replace("file://", "");
  const resumeDir = dirname(dirname(scriptPath));
  Deno.chdir(resumeDir);
}

// Open the PDF file in the default viewer (cross-platform)
console.log("Opening will_cygan_resume.pdf...");

const osType = Deno.build.os;
try {
  if (osType === "darwin") {
    // macOS
    await $`open will_cygan_resume.pdf`;
  } else if (osType === "linux") {
    // Linux
    await $`xdg-open will_cygan_resume.pdf`;
  } else if (osType === "windows") {
    // Windows
    await $`cmd /c start will_cygan_resume.pdf`;
  } else {
    console.log("Warning: Unknown OS type. Attempting to use xdg-open...");
    const result = await $`xdg-open will_cygan_resume.pdf`.noThrow();
    if (result.code !== 0) {
      console.log("Could not open PDF automatically");
    }
  }
} catch (error) {
  console.error("Error opening PDF:", error.message);
}

// Start typst watch in the foreground
console.log("Starting typst watch...");
await $`typst watch will_cygan_resume.typ`;

File: scripts/dev.py
#!/usr/bin/env -S uv run --script
# /// script
# requires-python = ">=3.11"
# dependencies = []
# ///
"""Opens the resume PDF and starts `typst watch`."""

from __future__ import annotations

import os
import subprocess
import sys
from pathlib import Path

RESUME_DIR = Path(__file__).resolve().parent.parent
sys.path.insert(0, str(RESUME_DIR))

from resume_tools import artifact  # noqa: E402
REQUEST = artifact.request_for_resume()
PDF = REQUEST.output.name


def open_pdf() -> None:
    print(f"Opening {PDF}...")
    try:
        if sys.platform == "darwin":
            subprocess.run(["open", PDF], check=False)
        elif sys.platform.startswith("linux"):
            subprocess.run(["xdg-open", PDF], check=False)
        elif sys.platform == "win32":
            os.startfile(PDF)  # type: ignore[attr-defined]
        else:
            print("Warning: unknown OS; trying xdg-open...")
            subprocess.run(["xdg-open", PDF], check=False)
    except FileNotFoundError as e:
        print(f"Could not open PDF automatically: {e}")


def main() -> int:
    os.chdir(RESUME_DIR)
    open_pdf()
    print("Starting typst watch...")
    return artifact.watch_artifact(REQUEST)


if __name__ == "__main__":
    raise SystemExit(main())
